### ISSUE TITLE
Block level disable comments depend on function formatting

### ISSUE THREAD

**Issue by AdamGleave** [PyCQA/pylint#3888]
### Steps to reproduce

Run `pylint` on the following code:

```python
"""Test cases."""

def f(
):  # pylint: disable=unused-variable
    # works
    i_am_unused = 42

def g():  # pylint:disable=unused-variable
    # fails
    i_am_unused = 42

def h(
) -> None:  # pylint:disable=unused-variable
    # fails
    i_am_unused = 42
```

### Current behavior

`pylint` honors the disable hint for `f`, but ignores it for `g` and `h`:

```
$ pylint test.py
************* Module test
test.py:10:4: W0612: Unused variable 'i_am_unused' (unused-variable)
test.py:15:4: W0612: Unused variable 'i_am_unused' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 5.00/10, +1.67)
```

### Expected behavior

`pylint` should honor the hint for all functions. (Or possibly none, if a function is not intended to count as a "block".) It should not depend on how the function signature is formatted.

### pylint --version output

```
$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.7.9 (default, Aug 18 2020, 02:07:21) 
[GCC 9.3.0]
```

**Comment by hippo91** [comment]
@AdamGleave thanks for this report. I can reproduce it.
